sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+
+        // customized pipeline
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+
+        TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new TextAnalyticsAsyncClient(textAnalyticsAPI, serviceVersion);
+    }
+
+    /**
+     * Set the default client option for one client.
+     *
+     * @param clientOptions TextAnalyticsClientOptions model that includes
+     * {@link TextAnalyticsClientOptions#getDefaultLanguage() default language} and
+     * {@link TextAnalyticsClientOptions#getDefaultCountryHint() default country hint}
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder clientOptions(TextAnalyticsClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Text Analytics instance.
+     *
+     * @param endpoint The URL of the Azure Text Analytics instance service requests to and receive responses from.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
+        }
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests for this TextAnalyticsClientBuilder.
+     *
+     * @param subscriptionKey subscription key
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException If {@code subscriptionKey} is {@code null}.
+     */
+    public TextAnalyticsClientBuilder subscriptionKey(String subscriptionKey) {
+        Objects.requireNonNull(subscriptionKey);
+
+        this.subscriptionKey = subscriptionKey;
+
+        // Clear TokenCredential in favor of subscription key credential
+        this.tokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential TokenCredential used to authenticate HTTP requests.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException If {@code tokenCredential} is {@code null}.
+     */
+    public TextAnalyticsClientBuilder credential(TokenCredential tokenCredential) {
+        // token credential can not be null value
+        Objects.requireNonNull(tokenCredential);
```

Comment:
For Objects.requireNonNull, you should have a message. Otherwise, all they see is a NullPointerException with no message.